src/compute-client/src/controller/instance.rs

```diff
@@ -679,6 +669,14 @@ impl<T: ComputeControllerTimestamp> Instance<T> {
         let mut history_updates = Vec::new();
         for (replica_id, replica) in &mut self.replicas {
             for (collection_id, collection) in &mut replica.collections {
+                if self
+                    .collections
+                    .get(collection_id)
+                    .expect("must exist")
+                    .read_only
+                {
+                    continue;
+                }
```

Comment:
I don't think this is what we want. We want to avoid recording wallclock lag measurements while the environment is in read-only mode during a 0dt upgrade, because they would be recorded in persist. But the new logic instead avoids recording measurements for read-only collections. Which works accidentally because if the environment is read-only, all collections are too. But once we introduce replacement MVs we'll want to have lag measurements for those, to monitor their progress/health.

I think we need to keep the `Instance::read_only` flag around, and differentiate between instance-level and collection-level read-only mode here.